opentracing/src/main/java/opentracing/propagation/SplitTextFormatInjector.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright 2016 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package opentracing.propagation;
+
+import java.util.Map;
+
+/**
+ *
+ */
+interface SplitTextFormatInjector extends Injector<Map> {
```

Comment:
if we're using `Map<String, String>` as the carrier type then there's no need for this to be part of `opentracing-java`... that said, there *does* need to be a comment somewhere about the carrier types (and semantics thereof) that `opentracing.Tracer` impls must support for injecting/joining.

Note that the other languages use two string maps to represent the "split" injected/extracted Span... in @adriancole's initial OT-java PR it looked like this: https://github.com/opentracing/opentracing-java/pull/4/files#diff-e4b78bf6849befce360697858137714dR37 (naming is OOD at this point, of course).

I think I prefer something like

```
interface SplitCarrier<E> {
  /** Implementation-specific Tracer State */
  void setTracerState(E);
  E getTracerState();

  /** Any Baggage that needs to be injected/extracted between processes */
  void setBaggage(E);
  E getBaggage();
}
```

(Or maybe even a concrete class rather than an interface; it's really just a struct, right?)

Then the aforementioned comment about mandatory carrier types would basically boil down to `SplitCarrier<Map<String, String>>` and `SplitCarrier<byte[]>`.